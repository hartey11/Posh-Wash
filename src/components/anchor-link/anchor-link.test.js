import React from "react"
import { render, cleanup } from "@testing-library/react"
import AnchorLink from "./anchor-link.component"

afterEach(cleanup)

const title = "Some Title"
const internalLink = "/this should encode correctly"
const externalLink = "https://www.bbc.co.uk/this should encode correctly"

describe("External Link", () => {
  it("should render correctly in the browser", () => {
    const { asFragment } = render(
      <AnchorLink title={title} href={externalLink}>
        Test
      </AnchorLink>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})

describe("Internal Link", () => {
  it("should render correctly in the browser", () => {
    const { asFragment } = render(
      <AnchorLink title={title} href={internalLink}>
        Test
      </AnchorLink>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
