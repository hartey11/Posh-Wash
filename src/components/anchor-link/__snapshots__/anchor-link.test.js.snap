// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`External Link should render correctly in the browser 1`] = `
<DocumentFragment>
  <a
    href="https://www.bbc.co.uk/this%20should%20encode%20correctly"
    rel="noopener noreferrer"
    target="_blank"
    title="Some Title"
  >
    Test
  </a>
</DocumentFragment>
`;

exports[`Internal Link should render correctly in the browser 1`] = `
<DocumentFragment>
  <a
    href="/this%20should%20encode%20correctly"
    title="Some Title"
  >
    Test
  </a>
</DocumentFragment>
`;
